meteor/server/api/ingest/syncChangesToPartInstance.ts

```diff
@@ -24,48 +24,66 @@ import {
 import { isTooCloseToAutonext, updateExpectedDurationWithPrerollForPartInstance } from '../playout/lib'
 import { CacheForIngest } from './cache'
 
+type PlayStatus = 'previous' | 'current' | 'next'
+type ReadOnlyIngestCacheWithoutRundown = Omit<ReadOnlyCache<CacheForIngest>, 'Rundown'>
+type SyncedInstance = {
+	existingPartInstance: PartInstance
+	previousPartInstance: PartInstance | undefined
+	playStatus: PlayStatus
+	newPart: Part
+	piecesThatMayBeActive: Promise<Piece[]>
+}
+
 export async function syncChangesToPartInstances(
 	cache: CacheForPlayout,
-	ingestCache: Omit<ReadOnlyCache<CacheForIngest>, 'Rundown'>,
+	ingestCache: ReadOnlyIngestCacheWithoutRundown,
 	showStyle: ReadonlyDeep<ShowStyleCompound>,
 	blueprint: ReadonlyDeep<ShowStyleBlueprintManifest>,
 	rundown: ReadonlyDeep<Rundown>
 ): Promise<void> {
 	if (cache.Playlist.doc.activationId) {
 		if (blueprint.syncIngestUpdateToPartInstance) {
 			const playlistPartInstances = getSelectedPartInstancesFromCache(cache)
-			const instances: Array<{
-				existingPartInstance: PartInstance
-				previousPartInstance: PartInstance | undefined
-				playStatus: 'current' | 'next'
-				newPart: Part
-				piecesThatMayBeActive: Promise<Piece[]>
-			}> = []
+			const instances: SyncedInstance[] = []
 			if (playlistPartInstances.currentPartInstance) {
-				const newPart = cache.Parts.findOne(playlistPartInstances.currentPartInstance.part._id)
-				if (newPart) {
-					instances.push({
-						existingPartInstance: playlistPartInstances.currentPartInstance,
-						previousPartInstance: playlistPartInstances.previousPartInstance,
-						playStatus: 'current',
-						newPart: newPart,
-						piecesThatMayBeActive: fetchPiecesThatMayBeActiveForPart(cache, ingestCache, newPart),
-					})
+				findPartAndInsertToSyncedInstanceCandidates(
+					instances,
+					cache,
+					ingestCache,
+					playlistPartInstances.currentPartInstance,
+					playlistPartInstances.previousPartInstance,
+					'current'
+				)
+				// If the currentPartInstance is adlibbed we probably also need to find the earliest
+				// non-adlibbed Part within this segment and check it for updates too. It may have something
+				// changed (like timing) that will affect what's going on.
+				if (playlistPartInstances.currentPartInstance.orphaned === 'adlib-part') {
```

Comment:
I think this needs to be done before the `findPartAndInsertToSyncedInstanceCandidates ` for `currentPartInstance`, because the input to the currentPartInstance will depend on the pieceInstances the previousPartInstance has, so this order will cause some changes to behave weirdly

Reply:
Didn't thought of that. Moved the "previous" Part Instance to be the first in the list.